chakrashim: re-enable chakra-config

jianchun opened the pull request:
Change chakracore.gyp to re-enable "chakra-config" variable.

By default "chakra-config" is hardcoded to "Release".

If a user needs to use Debug chakracore build for debugging purpose, flip
that variable value to "Debug".

(This was recently disabled to work around build cache issues. Seems the
issues were local. The end-to-end build experience is not great currently,
may need more work to improve.)

digitalinfinity commented on deps/chakrashim/chakracore.gyp:
```diff
@@ -7,11 +7,10 @@
-    'chakra_build_flags': '',
 
     # xplat (non-win32) only
     'chakra_config': 'Release',     # Debug, Release
```
Did you mean to change this to Release?

jianchun replied:
> Did you mean to change this to Release?

Yes it is "Release" by default (unchanged as before). What change do you mean?

jianchun replied:
> Also, should we have Test build support for node-chakra?

We could, never tried. Probably a good feature if we can also get "jscript.config" working. Otherwise not very useful if we can't pass flags.

digitalinfinity replied:
Sorry, meant to say - did you mean to change it to Debug? From your PR comment, it sounded like you were going to change the default too but guess not?

digitalinfinity commented on deps/chakrashim/chakracore.gyp:
```diff
@@ -7,11 +7,10 @@
-    'chakra_build_flags': '',
 
     # xplat (non-win32) only
     'chakra_config': 'Release',     # Debug, Release
```
Also, should we have Test build support for node-chakra?

kunalspathak replied:
@digitalinfinity , I don't feel a need for test build support so far. So not worth while adding it. 
FYI - This was the change I needed to make my `gdb` show locals.

digitalinfinity replied:
@jianchun are test builds of ChakraCore built with -g these days? I had a change locally that did that but I don't remember if I ever merged it...I don't think release builds are built with -g right now.

Commit:
chakrashim: re-enable chakra-config

Change chakracore.gyp to re-enable "chakra-config" variable.

By default "chakra-config" is hardcoded to "Release".

If a user needs to use Debug chakracore build for debugging purpose, flip
that variable value to "Debug".

(This was recently disabled to work around build cache issues. Seems the
issues were local. The end-to-end build experience is not great currently,
may need more work to improve.)

PR-URL: https://github.com/nodejs/node-chakracore/pull/135
Reviewed-By: Kunal Pathak <Kunal.Pathak@microsoft.com>
Reviewed-By: Hitesh Kanwathirtha <hiteshk@microsoft.com>
